crates/typst/src/layout/grid/layout.rs

```diff
@@ -139,19 +215,54 @@ impl Entry {
     }
 }
 
+/// A grid item, possibly affected by automatic cell positioning. Can be either
+/// a line or a cell.
+pub enum GridItem<T: ResolvableCell> {
+    /// A horizontal line in the grid.
+    HLine {
```

Comment:
I thought about it, but I didn't see much value at first since we will already be converting to the unified `Line` struct immediately anyway. If this sounds better or otherwise more consistent to you, it's certainly doable though.

Reply:
To provide a bit of further context, I was walking in that direction when I made the `pending_hlines` thing, since it required pushing lines to a vector and later extracting information from them, which is very inconvenient without a dedicated struct. However, turns out I could just create a `Line` right away anyway and only have to match on the GridItem enum once. So I think it should be fine to keep as is, unless we happen to face a situation where we'd need to match twice or more on GridItem in the future.

Reply:
Okay, makes sense.